From: "Rémi Bernon" <rbernon@codeweavers.com>
Subject: [PATCH 5/5] ntoskrnl.exe: Make user shared data pointers volatile (GCC 11).
Message-Id: <20210927085833.544992-5-rbernon@codeweavers.com>
Date: Mon, 27 Sep 2021 10:58:33 +0200
In-Reply-To: <20210927085833.544992-1-rbernon@codeweavers.com>
References: <20210927085833.544992-1-rbernon@codeweavers.com>

So that GCC 11 stops warning about reading from a 0-size memory region.

Signed-off-by: Rémi Bernon <rbernon@codeweavers.com>
---
 dlls/ntoskrnl.exe/instr.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/dlls/ntoskrnl.exe/instr.c b/dlls/ntoskrnl.exe/instr.c
index 11111111111..11111111111 100644
--- a/dlls/ntoskrnl.exe/instr.c
+++ b/dlls/ntoskrnl.exe/instr.c
@@ -496,8 +496,8 @@ WINE_DEFAULT_DEBUG_CHANNEL(int);
 #define SIB_BASE( sib, rex )    (((sib) & 7) | (((rex) & REX_B) ? 8 : 0))
 
 /* keep in sync with dlls/ntdll/thread.c:thread_init */
-static const BYTE *wine_user_shared_data = (BYTE *)0x7ffe0000;
-static const BYTE *user_shared_data      = (BYTE *)0xfffff78000000000;
+static const BYTE *const volatile wine_user_shared_data = (BYTE *)0x7ffe0000;
+static const BYTE *const volatile user_shared_data      = (BYTE *)0xfffff78000000000;
 
 static inline DWORD64 *get_int_reg( CONTEXT *context, int index )
 {

-- 
0.0.0

